src/OrchardCore.Modules/OrchardCore.Media.Azure/Services/MediaBlobFileStoreVersionProvider.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Options;
+using OrchardCore.FileStorage;
+using OrchardCore.FileStorage.AzureBlob;
+
+namespace OrchardCore.Media.Azure.Services
+{
+    public class MediaBlobFileStoreVersionProvider : IFileStoreVersionProvider
+    {
+        private const string VersionKey = "v";
+
+        private readonly IMediaFileStore _mediaFileStore;
+        private readonly IMemoryCache _cache;
+        private readonly MediaBlobStorageOptions _options;
+
+        public MediaBlobFileStoreVersionProvider(
+            IMediaFileStore mediaFileStore,
+            IMemoryCache cache,
+            IOptions<MediaBlobStorageOptions> options
+            )
+        {
+            _mediaFileStore = mediaFileStore;
+            _cache = cache;
+            _options = options.Value;
+        }
+
+        public async Task<string> AddFileVersionToPathAsync(string resolvedPath, string path)
+        {
+            var cacheKey = resolvedPath;
+
+            // Path has already been correctly parsed before here.
+            resolvedPath = _mediaFileStore.MapPublicUrlToPath(resolvedPath);
+
+            var fileInfo = await _mediaFileStore.GetFileInfoAsync(resolvedPath) as BlobFile;
+
+            if (fileInfo != null)
+            {
+                //TODO Consider expiring with a Change Token from Blob File Store.
+                // Or consider a graceful cache refresh.
+                var cacheEntryOptions = new MemoryCacheEntryOptions()
+                    .SetAbsoluteExpiration(
+                        TimeSpan.FromMinutes(_options.VersionHashCacheExpiryTime > 0 ? _options.VersionHashCacheExpiryTime : 120))
```

Comment:
Isn't the solution to set `120` as a default value in the class? And then the options would be reset if specified. but I might be wrong and we need to check with the media file types which have the same pattern.

Reply:
Yes, my mistake leaving this in, so removed, and relying on defaults.

**Note** for file types, and supported sizes, being arrays, they're handled (as key/value pairs) additively by `.Configure<IOptions<T>>` , so I had to use `IConfigureOptions` to configure them, otherwise the defaults would add to the `appsettings.json` value, rather than replacing it.